Sources/MongoSwift/MongoDatabase.swift

```diff
@@ -478,6 +478,79 @@ public struct MongoDatabase {
+     *   - options: Optional `AggregateOptions` to use when executing the command
+     *   - session: Optional `ClientSession` to use when executing this command
+     *
+     * - Warning:
+     *    If the returned cursor is alive when it goes out of scope, it will leak resources. To ensure the cursor
+     *    is dead before it leaves scope, invoke `MongoCursor.kill(...)` on it.
+     *
+     * - Returns:
+     *    An `EventLoopFuture<MongoCursor>`. On success, contains a cursor over the resulting documents.
+     *
+     *    If the future fails, the error is likely one of the following:
+     *    - `MongoError.InvalidArgumentError` if the options passed are an invalid combination.
+     *    - `MongoError.LogicError` if the provided session is inactive.
+     *    - `MongoError.LogicError` if this database's parent client has already been closed.
+     *    - `EncodingError` if an error occurs while encoding the options to BSON.
+     *
+     * - SeeAlso: https://docs.mongodb.com/manual/reference/method/db.aggregate/
+     */
+    public func aggregate(
+        _ pipeline: [BSONDocument],
+        options: AggregateOptions? = nil,
+        session: ClientSession? = nil
+    ) -> EventLoopFuture<MongoCursor<BSONDocument>> {
+        self.aggregate(pipeline, options: options, session: session, withOutputType: BSONDocument.self)
+    }
+
+    /**
+     * Runs an aggregation framework pipeline against this database for pipeline stages that do not require an
+     * underlying collection, such as `$currentOp` and `$listLocalSessions`.
+     * Associates the specified `Codable` type `OutputType` with the returned `MongoCursor`
+     *
+     * - Parameters:
+     *   - pipeline: an `[Document]` containing the pipeline of aggregation operations to perform.
+     *   - options: Optional `AggregateOptions` to use when executing the command
+     *   - session: Optional `ClientSession` to use when executing this command
+     *   - withOutputType: the type that each resulting document of the output
+     *     of the aggregation operation will be decoded to
+     *
+     * - Warning:
+     *    If the returned cursor is alive when it goes out of scope, it will leak resources. To ensure the cursor
+     *    is dead before it leaves scope, invoke `MongoCursor.kill(...)` on it.
+     *
+     * - Returns:
+     *    An `EventLoopFuture<MongoCursor>`over the resulting `OutputType`s
+     *
+     *    If the future fails, the error is likely one of the following:
+     *    - `MongoError.InvalidArgumentError` if the options passed are an invalid combination.
+     *    - `MongoError.LogicError` if the provided session is inactive.
+     *    - `MongoError.LogicError` if this database's parent client has already been closed.
+     *    - `EncodingError` if an error occurs while encoding the options to BSON.
+     *
+     * - SeeAlso: https://docs.mongodb.com/manual/reference/method/db.aggregate/
```

Comment:
We try to avoid linking directly to the shell's documentation in drivers docs since it can confuse users when the shell adopts an API different from the one adopted by the drivers. I think we could instead link to https://docs.mongodb.com/manual/reference/operator/aggregation-pipeline/#db-aggregate-stages to list the stages supported by db-level aggregation. Unfortunately this page is littered with links to the shell documentation too, but I think it's still an improvement over us linking directly to shell documentation.